### PR TITLE
Sagemaker endpoint config kms key configured

### PR DESCRIPTION
**Pull request by Geeth26**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py]: Remove unnecessary comments

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    if not rule_parameters:
+        return {}
+
+    if not 'keyIds' in rule_parameters:
+        return {}
+
+    given_keyids = rule_parameters['keyIds'].replace(' ', '').split(',')
+    valid_keyids = []
+
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py]: initialise this at the start of the function and return it as it is, i.e. blank:
```
valid_key_ids = []
if "keyArns" in rule_parameters:
    ...
    valid_key_ids.append()
    ...
return valid_key_ids
```

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    rule_parameters_scenario5 = '{"keyIds":"arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d, arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3f"}'
+    list_endpoints_scenario5 = [{'EndpointConfigs':[{'EndpointConfigName':'endpoint1'}, {'EndpointConfigName':'endpoint2'}]}]
+    described_endpoints_scenario5 = [{'EndpointConfigName':'endpoint1', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint1'}, {'EndpointConfigName': 'endpoint2', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint2', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3f'}]
+
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: The same rule parameters and list_endpoints can be used across scenarios 3, 4 and 5. Just use modified described_endpoints to vary compliance scenarios.

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    rule_parameters_scenario6 = '{}'
+    list_endpoints_scenario6 = [{'EndpointConfigs':[{'EndpointConfigName':'endpoint1'}, {'EndpointConfigName':'endpoint2'}]}]
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: same list_endpoints can be used here as above

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    described_endpoints_scenario6 = [{'EndpointConfigName':'endpoint1', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint1'}, {'EndpointConfigName': 'endpoint2', 'EndpointConfigArn': 'arn:aws:sagemaker:us-east-1:305333956852:endpoint-config/endpoint2', 'KmsKeyId': 'arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3f'}]
+
+
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: Remove extra blank line

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    invoking_event_iam_role_sample = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"some-resource-name","ARN":"some-arn"},"notificationCreationTime":"2018-07-02T23:05:34.445Z","messageType":"ConfigurationItemChangeNotification"}'
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: Not used anywhere. Remove this variable

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+            "paginate.return_value": self.list_endpoints_scenario3})
+        lambda_event = build_lambda_scheduled_event(rule_parameters=self.rule_parameters_scenario3)
+        SAGEMAKER_CLIENT_MOCK.describe_endpoint_config = MagicMock(side_effect=self.described_endpoints_scenario3)
+        response = RULE.lambda_handler(lambda_event, {})
+        resp_expected = []
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'endpoint1', annotation="No AWS KMS Key is configured for this Amazon SageMaker Endpoint Config."))
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'endpoint2', annotation="No AWS KMS Key is configured for this Amazon SageMaker Endpoint Config."))
+        assert_successful_evaluation(self, response, resp_expected, 2)
+
+    def test_scenario_4_no_matching_keyids(self):
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: comments for scenarios

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        assert_successful_evaluation(self, response, resp_expected, 2)
+
+    def test_scenario_6_compliant(self):
+        SAGEMAKER_CLIENT_MOCK.configure_mock(**{
+            "get_paginator.return_value": SAGEMAKER_CLIENT_MOCK,
+            "paginate.return_value": self.list_endpoints_scenario6})
+        SAGEMAKER_CLIENT_MOCK.describe_endpoint_config = MagicMock(side_effect=self.described_endpoints_scenario6)
+        lambda_event = build_lambda_scheduled_event(rule_parameters=self.rule_parameters_scenario6)
+        resp_expected = []
+        response = RULE.lambda_handler(lambda_event, {})
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: move line to above "resp_expected = []"

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+        resp_expected.append(build_expected_response('COMPLIANT', 'endpoint1'))
+        resp_expected.append(build_expected_response('COMPLIANT', 'endpoint2'))
+        assert_successful_evaluation(self, response, resp_expected, 2)
+
+class ParametersTest(unittest.TestCase):
+    rule_parameters_scenario1 = '{"keyIds":"arn:aws:kms:us-east-1:305333956852:key/ae25566a-c0d4-4ed2-8131-7c98e9487e3d, arn:als:kms:us-east-1:305333956852:keys/ae25566a-c0d4-4ed2-8131-7c98e9487e3d"}'
+
+    def test_scenario1(self):
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED_test.py]: comments for scenarios

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py (hunk 2 above)
Comment (kritijha) [reply python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py]: @Geeth26 The evaluate_parameters method needs to be modified as per above comment

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py [hunk 10]
```diff
@@ -0,0 +1,458 @@
...
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+def get_all_endpoint_config_names(client):
+    """This function returns all the Sage maker endpoint configs """
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.py]: Sage maker endpoint configs -> SageMaker Endpoint Configs

**Comment by jongogogo** [comment]
When the update is planned?

**Comment by kritijha** [comment]
@Geeth26 As discussed offline, please update the PR with all the changes.

File: python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/parameters.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
+  "Version": "1.0", 
+  "Parameters": {
+    "CodeKey": "SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED.zip", 
+    "SourceRuntime": "python3.6", 
+    "SourcePeriodic": "One_Hour", 
+    "RuleName": "SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED", 
+    "OptionalParameters": "{}",
```
Comment (kritijha) [inline python/SAGEMAKER_ENDPOINT_CONFIG_KMS_KEY_CONFIGURED/parameters.json]: The optional parameter should contain "keyArns"

**Comment by kritijha** [comment]
Thanks @Geeth26 for the updated PR! Great work :)
@jongogogo please review!

**Comment by jongogogo** [comment]
You deleted the gitignore in your PR, please don't

**Comment by jongogogo** [comment]
The rest is good, just update your PR without deleting the gitgnore 👍

**Pull request merged**